xgi/core/hypergraph.py

```diff
@@ -1462,49 +1462,38 @@ def cleanup(
 
         """
         if in_place:
-            if not multiedges:
-                self.merge_duplicate_edges()
-            if not singletons:
-                self.remove_edges_from(self.edges.singletons())
-            if not isolates:
-                self.remove_nodes_from(self.nodes.isolates())
-            if connected:
-                from ..algorithms import largest_connected_component
-
-                self.remove_nodes_from(self.nodes - largest_connected_component(self))
-            if relabel:
-                from ..utils import convert_labels_to_integers
-
-                temp = convert_labels_to_integers(self).copy()
-
-                nn = temp.nodes
-                ee = temp.edges
-
-                self.clear()
-                self.add_nodes_from((n, deepcopy(attr)) for n, attr in nn.items())
-                self.add_edges_from(
-                    (e, id, deepcopy(temp.edges[id]))
-                    for id, e in ee.members(dtype=dict).items()
-                )
-                self._net_attr = deepcopy(temp._net_attr)
+            _H = self
         else:
-            H = self.copy()
-            if not multiedges:
-                H.merge_duplicate_edges()
-            if not singletons:
-                H.remove_edges_from(H.edges.singletons())
-            if not isolates:
-                H.remove_nodes_from(H.nodes.isolates())
-            if connected:
-                from ..algorithms import largest_connected_component
-
-                H.remove_nodes_from(H.nodes - largest_connected_component(H))
-            if relabel:
-                from ..utils import convert_labels_to_integers
-
-                H = convert_labels_to_integers(H)
-
-            return H
+            _H = self.copy()
+
+        if not multiedges:
+            _H.merge_duplicate_edges()
+        if not singletons:
+            _H.remove_edges_from(_H.edges.singletons())
+        if not isolates:
+            _H.remove_nodes_from(_H.nodes.isolates())
+        if connected:
+            from ..algorithms import largest_connected_hypergraph
+
+            largest_connected_hypergraph(_H, in_place=True)
+        if relabel:
+            from ..utils import convert_labels_to_integers
+
+            temp = convert_labels_to_integers(_H).copy()
+
+            nn = temp.nodes
+            ee = temp.edges
+
+            _H.clear()
+            _H.add_nodes_from((n, deepcopy(attr)) for n, attr in nn.items())
+            _H.add_edges_from(
+                (e, id, deepcopy(temp.edges[id]))
```

Comment:
can use ee instead of temp.edges

Reply:
Good catch! I have fixed this and moved all of the heavy lifting to the `convert_labels_to_integers` function.